examples/systemcalls/systemcalls.c

```diff
@@ -39,15 +72,48 @@ bool do_exec(int count, ...)
     va_list args;
     va_start(args, count);
     char * command[count+1];
+    char * rest_of_args[count];
     int i;
     for(i=0; i<count; i++)
     {
         command[i] = va_arg(args, char *);
     }
+    
+    // checks if directory exists or not
+    DIR* dir = opendir(command[0]);
+    if(dir)
+    {
+    	closedir(dir);
+    }
+    else if(errno == ENOENT)
+    {
+    	return false;
+    }
+    
+    // add situation if number of arguments is 3
+    if(count == 3)
+    {
+    	DIR* dir1 = opendir(command[2]);
+    	if(dir1)
+    	{
+    		closedir(dir1);
+    	}
+    	else if(errno == ENOENT)
+    	{
+    		return false;
+    	}
+    }
+    // assign full path variable
+    char *full_path = command[0];
     command[count] = NULL;
     // this line is to avoid a compile warning before your implementation is complete
     // and may be removed
     command[count] = command[count];
+    
+    for(i=0; i<(count+1); i++)
+    {
+        rest_of_args[i]=command[i+1];
+    }
```

Comment:
@abaa3023 I don't think this is necessary, you can just use `execv(command[0], command)`